client/dashboard/domains/dns/edit.tsx

```diff
@@ -0,0 +1,73 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { useNavigate } from '@tanstack/react-router';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { domainDnsMutation, domainDnsQuery } from '../../app/queries/domain-dns-records';
+import { domainRoute } from '../../app/routes/domain-routes';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import DNSRecordForm from './form';
+import { DNS_RECORD_CONFIGS } from './records/dns-record-configs';
+import type { DnsRecordTypeFormData, DnsRecordFormData } from './records/dns-record-configs';
+import type { DnsRecord } from '../../data/domain-dns-records';
+
+export default function DomainEditDNS() {
+	const navigate = useNavigate();
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const { domainName } = domainRoute.useParams();
+	const { recordId } = domainRoute.useSearch();
+	const mutation = useMutation( domainDnsMutation( domainName ) );
+
+	const { data: dnsRecords } = useQuery( domainDnsQuery( domainName ) );
```

Comment:
Since we've loaded the `domainDnsQuery` in the router, we can use suspense here.
```suggestion
	const { data: dnsRecords } = useSuspenseQuery( domainDnsQuery( domainName ) );
```

Which is cool because the TypeScript type for `dnsRecords` will now never by `undefined`

Reply:
This is nice! I'm learning a lot about Tanstack and TypeScript with these reviews 😄